pkg/operator/sync.go

```diff
@@ -1417,10 +1416,26 @@ func (optr *Operator) reconcileSimpleContentAccessSecrets(layeredMCPs []*mcfgv1.
 		secretName := ctrlcommon.SimpleContentAccessSecretName + "-" + poolName
 
 		// Create a clone of simpleContentAccessSecret, modify it to be in the MCO namespace with the new name
+		currentPool, err := optr.mcpLister.Get(poolName)
+		if err != nil {
+			return err
+		}
+
+		// Work around https://github.com/kubernetes/kubernetes/issues/3030 and https://github.com/kubernetes/kubernetes/issues/80609
+		currentPool.APIVersion = mcfgv1.GroupVersion.String()
+		currentPool.Kind = "MachineConfigPool"
 		clonedSecret := &corev1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      secretName,
 				Namespace: ctrlcommon.MCONamespace,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: currentPool.APIVersion,
+						Kind:       currentPool.Kind,
+						Name:       currentPool.ObjectMeta.Name,
+						UID:        currentPool.ObjectMeta.UID,
+					},
+				},
```

Comment:
This was a minor fix for a particular scenario I encountered during e2es: If the pool was deleted prior to being opted out of layering, this sync function wouldn't delete it. Adding the MCP as an owner field seemed like the cleanest fix.